Assets/Scripts/CharacterControllers/Melody/MelodyController.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace Melody
+{
+    using HarmonyQuest;
+    using HarmonyQuest.Input;
+    using Melody.States;
+    using UnityEngine;
+
+    public class MelodyController : MonoBehaviour
+    {
+        readonly CharacterController characterController;
+        MelodyStateMachine StateMachine;
+
+        Vector3 move;
```

Comment:
I'll nitpick the lack of private statements on these, but I did some reading and apparently C#'s Default member accessibility for classes is private